internal/test/testing.go

```diff
@@ -54,10 +57,192 @@ func NewConnection(api auth.Api, tokenVersion auth.TokenVersion) *Connection {
 		DomainName: tenantDomain,
 	}
 
-	t.Authorizer, err = t.AuthConfig.NewAuthorizer(t.Context, api)
+	return &t
+}
+
+// Authorize configures an Authorizer for the Connection
+func (c *Connection) Authorize(api environments.Api) {
+	var err error
+	c.Authorizer, err = c.AuthConfig.NewAuthorizer(c.Context, api)
```

Comment:
this should pass the context onto the methods rather than being on the Connection object here?

Reply:
Given this helper is for acceptance testing, I went with moving the context into the top-level `test.Test` struct that gets instantiated at the start of each test, then ensured the method on the `Connection` struct method accepts a ctx argument, and ensured the cancel func is deferred for every test. Although in testing I noticed that the context in `testing.T` seems to always return an empty deadline anyway!